### PR TITLE
copr: don't convert bool to int while comparing json bool with other types

### LINKED ISSUE #13386: keep json comparison compatible with TiDB and MySQL
## Development Task

Fix https://github.com/pingcap/tidb/issues/37481, to keep the comparison of bool and other json types the same with TiDB and MySQL.

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: YangKeao <yangkeao@chunibyo.icu>

### What is changed and how it works?

Issue Number: Close #13386

What's Changed:

Remove the special statements to convert the json bool to int and comparing them. It fixes the https://github.com/pingcap/tidb/issues/37481

### Check List 

Tests 

- [x] Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

### Release note 

```release-note
Fix the issue that comparison between json bool and other json value is not compatible with TiDB and MySQL
```

### CODE REVIEW

**Comment by YangKeao** [comment]
@xiongjiwei PTAL

**Comment by YangKeao** [comment]
@xiongjiwei PTAL

**Comment by YangKeao** [comment]
@xiongjiwei PTAL

**Comment by YangKeao** [comment]
/cc @wshwsh12 @gengliqi @sticnarf 

PTAL :heart:

File: components/tidb_query_datatype/src/codec/datum.rs [hunk 1]
```diff
@@ -1960,7 +1960,7 @@ mod tests {
             ),
             (Datum::Bytes(b"[1, 2, 3]".to_vec()), "[1, 2, 3]"),
             (Datum::Bytes(b"{}".to_vec()), "{}"),
-            (Datum::I64(1), "true"),
```
Comment (sticnarf) [inline components/tidb_query_datatype/src/codec/datum.rs:1963]: Why it even affects cast...

Comment (YangKeao) [reply components/tidb_query_datatype/src/codec/datum.rs:1963]: It doesn't, but this test verifies whether the output is expected through `assert_eq :facepalm: .

**Review by sticnarf — APPROVED** [review]

**Review by xiongjiwei — APPROVED** [review]

**Review by gengliqi — APPROVED** [review]

**Pull request merged**